components/insight/SRC/org/openmicroscopy/shoola/agents/fsimporter/util/LightFileImportComponent.java

```diff
@@ -0,0 +1,1154 @@
+            resultIndex = ImportStatus.STARTED;
+            Status sl = (Status) evt.getNewValue();
+            if (sl.equals(status)) {
+                firePropertyChange(Status.FILE_IMPORT_STARTED_PROPERTY, null,
+                        this);
+            }
+        } else if (Status.UPLOAD_DONE_PROPERTY.equals(name)) {
+            Status sl = (Status) evt.getNewValue();
+            if (sl.equals(status) && hasParent()) {
+                if (sl.isMarkedAsCancel())
+                    cancel(true);
+                else {
+                    formatResult();
+                    firePropertyChange(Status.UPLOAD_DONE_PROPERTY, null, this);
+                }
+            }
+        } else if (Status.FILE_RESET_PROPERTY.equals(name)) {
+            importable.setFile((File) evt.getNewValue());
+        } else if (ThumbnailLabel.BROWSE_PLATE_PROPERTY.equals(name)) {
+            firePropertyChange(BROWSE_PROPERTY, evt.getOldValue(),
+                    evt.getNewValue());
+        } else if (Status.CONTAINER_FROM_FOLDER_PROPERTY.equals(name)) {
+            containerFromFolder = (DataObject) evt.getNewValue();
+            if (containerFromFolder instanceof DatasetData) {
+                containerObject = containerFromFolder;
+            } else if (containerFromFolder instanceof ScreenData) {
+                containerObject = containerFromFolder;
+            }
+        } else if (Status.DEBUG_TEXT_PROPERTY.equals(name)) {
+            firePropertyChange(name, evt.getOldValue(), evt.getNewValue());
+        } else if (ThumbnailLabel.VIEW_IMAGE_PROPERTY.equals(name)) {
+            // use the group
+            SecurityContext ctx = new SecurityContext(importable.getGroup()
+                    .getId());
+            EventBus bus = ImporterAgent.getRegistry().getEventBus();
+            Long id = (Long) evt.getNewValue();
+            bus.post(new ViewImage(ctx, new ViewImageObject(id), null));
+        } else if (Status.IMPORT_DONE_PROPERTY.equals(name)
+                || Status.PROCESSING_ERROR_PROPERTY.equals(name)) {
+            Status sl = (Status) evt.getNewValue();
+            if (sl.equals(status))
+                firePropertyChange(Status.IMPORT_DONE_PROPERTY, null, this);
+        }
+        else if (Status.STEP_PROPERTY.equals(name)) {
+            firePropertyChange(Status.STEP_PROPERTY, evt.getOldValue(), evt.getNewValue());
+        }
+    }
+
+    /*
+     * (non-Javadoc)
+     * 
+     * @see
+     * org.openmicroscopy.shoola.agents.fsimporter.util.FileImportComponentI
+     * #toString()
+     */
+    @Override
+    public String toString() {
+        StringBuffer buf = new StringBuffer();
+        buf.append(getFile().getAbsolutePath());
+        if (importable.getGroup() != null)
+            buf.append("_" + importable.getGroup().getId());
+        if (importable.getUser() != null)
+            buf.append("_" + importable.getUser().getId());
```

Comment:
If there's either a group or a user we can't tell which kind of ID it is?

Reply:
I remove the toString methods, guess they were used for debug purposes once.